projects/components/src/draggable-list/draggable-list.component.ts

```diff
@@ -0,0 +1,49 @@
+import { CdkDragDrop, moveItemInArray } from '@angular/cdk/drag-drop';
+import {
+  AfterContentInit,
+  ChangeDetectionStrategy,
+  Component,
+  ContentChildren,
+  EventEmitter,
+  Input,
+  Output,
+  QueryList
+} from '@angular/core';
+import { DraggableItemComponent } from './draggable-item/draggable-item.component';
+
+@Component({
+  selector: 'ht-draggable-list',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div cdkDropList class="example-list" (cdkDropListDropped)="dropList($event)">
+      <div
+        *ngFor="let draggableItem of this.draggableItems"
+        cdkDrag
+        [cdkDragDisabled]="disabled || draggableItem.disabled"
+      >
+        <ng-container *ngTemplateOutlet="draggableItem.content"></ng-container>
+      </div>
+    </div>
+  `
+})
+export class DraggableListComponent implements AfterContentInit {
+  @Input()
+  public disabled: boolean = false;
+
+  @Output()
+  public readonly listOrderChange: EventEmitter<DraggableItemComponent[]> = new EventEmitter();
+
+  @ContentChildren(DraggableItemComponent)
+  public draggableItemsRef!: QueryList<DraggableItemComponent>;
```

Comment:
We might have to wrap this with `QueryListAndChanges`. Otherwise, it may not capture any dynamically added DraggableItemComponent

Reply:
@anandtiwary are you sure? If I add QueryListAndChanges will be triggered after call dropList and that would be wrong.